src/Nethermind/Nethermind.JsonRpc.TraceStore/TraceStoreRpcModule.cs

```diff
@@ -0,0 +1,311 @@
+        }
+
+        return _traceModule.trace_block(blockParameter);
+    }
+
+    public ResultWrapper<IEnumerable<ParityTxTraceFromStore>> trace_get(Keccak txHash, long[] positions)
+    {
+        ResultWrapper<IEnumerable<ParityTxTraceFromStore>> traceTransaction = trace_transaction(txHash);
+        List<ParityTxTraceFromStore> traces = TraceRpcModule.ExtractPositionsFromTxTrace(positions, traceTransaction);
+        return ResultWrapper<IEnumerable<ParityTxTraceFromStore>>.Success(traces);
+    }
+
+    public ResultWrapper<IEnumerable<ParityTxTraceFromStore>> trace_transaction(Keccak txHash) =>
+        TryTraceTransaction(
+            txHash,
+            ParityTraceTypes.Trace,
+            static t => ParityTxTraceFromStore.FromTxTrace(t),
+            out ResultWrapper<IEnumerable<ParityTxTraceFromStore>>? result)
+        && result is not null
+            ? result
+            : _traceModule.trace_transaction(txHash);
+
+    private bool TryTraceTransaction<T>(
+        Keccak txHash,
+        ParityTraceTypes traceTypes,
+        Func<ParityLikeTxTrace, T> map,
+        out ResultWrapper<T>? result)
+    {
+        SearchResult<Keccak> blockHashSearch = _receiptFinder.SearchForReceiptBlockHash(txHash);
+        if (blockHashSearch.IsError)
+        {
+            {
+                result = ResultWrapper<T>.Fail(blockHashSearch);
+                return true;
+            }
+        }
+
+        SearchResult<Block> blockSearch = _blockFinder.SearchForBlock(new BlockParameter(blockHashSearch.Object!));
+        if (blockSearch.IsError)
+        {
+            {
+                result = ResultWrapper<T>.Fail(blockSearch);
+                return true;
+            }
+        }
+
+        Block block = blockSearch.Object!;
+
+        if (TryGetBlockTraces(block.Header, out List<ParityLikeTxTrace>? traces) && traces is not null)
+        {
+            ParityLikeTxTrace? trace = GetTxTrace(block, txHash, traces);
+            if (trace is not null)
+            {
+                FilterTrace(trace, traceTypes);
+                result = ResultWrapper<T>.Success(map(trace));
+                return true;
+            }
+        }
+
+        result = null;
+        return false;
+    }
+
+    private bool TryGetBlockTraces(BlockHeader block, out List<ParityLikeTxTrace>? traces)
+    {
+        Span<byte> tracesSerialized = _traceStore.GetSpan(block.Hash!);
+        try
+        {
+            if (!tracesSerialized.IsEmpty)
+            {
+                List<ParityLikeTxTrace>? tracesDeserialized = _traceSerializer.Deserialize(tracesSerialized);
+                if (tracesDeserialized is not null)
+                {
+                    if (_logger.IsTrace) _logger.Trace($"Found persisted traces for block {block.ToString(BlockHeader.Format.FullHashAndNumber)}");
+                    traces = tracesDeserialized;
+                    return true;
+                }
+            }
+
+            traces = null;
+            return false;
```

Comment:
NP: I'm guessing returning trace directly and checking `traces is null` would also work.